stacker/actions/build.py

```diff
@@ -255,6 +255,7 @@ def _launch_stack(self, stack, **kwargs):
 
                 return FailedStatus(reason)
             elif self.provider.is_stack_completed(provider_stack):
+                self.provider.set_outputs(stack.fqn, provider_stack)
```

Comment:
This is a simple change to ensure that output lookups are parallelized. I'm not sure this is really the best implementation for this, which is why there's no comment explaining this yet.

In a nutshell, when a stack is "UPDATE_COMPLETE", this will store the outputs for the stack on the provider. When another stack references an output from a dependency, this ensures that there's no sequential blocking DescribeStacks calls to do the output lookups from dependencies, and also makes outputs thread safe.